source/DataWriter.h

```diff
@@ -33,86 +35,125 @@ class DataNode;
 // automatically adds quotation marks around strings if they contain whitespace.
 class DataWriter {
 public:
-	// Constructor, specifying the file to write.
-	explicit DataWriter(const std::string &path);
 	// Constructor for a DataWriter that will not save its contents automatically
 	DataWriter();
 	DataWriter(const DataWriter &) = delete;
 	DataWriter(DataWriter &&) = delete;
 	DataWriter &operator=(const DataWriter &) = delete;
 	DataWriter operator=(DataWriter &&) = delete;
-	// The file is not actually saved until the destructor is called. This makes
-	// it possible to write the whole file in a single chunk.
-	~DataWriter();
```

Comment:
I think we should keep this behavior, since it's pretty useful.

Reply:
Not anymore. You asked me to remove the SaveToPath etc. functions that would have done the work in that function (along with the variable that stored the file path). I guess I could revert that, but without those, it's useless.

Reply:
Why is it useless without those? But yeah you'd need need the file path variable.

Reply:
I guess I'll re-add SaveToPath and the orignal file path variable.